runtime/lib/ttnn/operations/include/tt/runtime/ttnn/operations/eltwise/binary/utils.cpp

```diff
@@ -0,0 +1,25 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+#include "utils.h"
+#include "tt/runtime/detail/logger.h"
+
+namespace tt::runtime::ttnn::operations::binary {
+
+void getEltwiseBinaryOPInputTensors(const ::tt::target::ttnn::EltwiseOp *op,
+                                    ProgramTensorPool &tensorPool,
+                                    ::ttnn::Tensor **lhs,
+                                    ::ttnn::Tensor **rhs) {
+  LOG_ASSERT(op->ins()->size() == 2, "Expected 2 inputs");
+  *lhs = &(tensorPool.at(op->ins()->Get(0)->global_id()));
+  *rhs = &(tensorPool.at(op->ins()->Get(1)->global_id()));
+  DEBUG_ASSERT((*lhs)->is_allocated());
+  DEBUG_ASSERT((*rhs)->is_allocated());
+
+  // Switch the order of operands if the second operand requires broadcast
+  if ((*rhs)->volume() < (*lhs)->volume()) {
```

Comment:
We should guard this underneath a workaround.

Reply:
Sounds good, FYI @uazizTT I think you added this, is this still required in runtime?

Reply:
Created issue #1124

Reply:
> Sounds good, FYI @uazizTT I think you added this, is this still required in runtime? Are there plans to do this in the compiler?

We don't intend to fix it in the compiler as it should be fixed in tt-metal, this is the reported issue in tt-metal to track it. https://github.com/tenstorrent/tt-mlir/issues/582

Once tt-metal fixes it on their end, this workaround can be removed, but until then it can be guarded as a workaround in runtime.